lib/resource-mapper.js

```diff
@@ -90,6 +90,14 @@ class ResourceMapper {
     return { path, contentType: contentType || this._defaultContentType }
   }
 
+  async _getMatchingFile (folder, filename, isIndex) {
+    const files = await this._readdir(folder)
+
+    // Find a file with the same name (minus the dollar extension)
+    return (files.find(f => this._removeDollarExtension(f) === filename ||
```

Comment:
We should probably expand this condition across multiple lines now, given the increased complexity.